src/DssTest.sol

```diff
@@ -323,6 +325,106 @@ abstract contract DssTest is Test {
         checkFileAddress(_base, _contractName, values);
     }
 
+    /// @dev This is forge-only due to event checking
+    function checkFileString(address _base, string memory _contractName, string[] memory _values) internal {
+        FileLike base = FileLike(_base);
+        uint256 ward = base.wards(address(this));
+
+        // Ensure we have admin access
+        GodMode.setWard(_base, address(this), 1);
+
+        // First check an invalid value
+        vm.expectRevert(abi.encodePacked(_contractName, "/file-unrecognized-param"));
+        base.file("an invalid value", "");
+
+        // Next check each value is valid and updates the target storage slot
+        for (uint256 i = 0; i < _values.length; i++) {
+            string memory value = _values[i];
+            bytes32 valueB32;
+            assembly {
+                valueB32 := mload(add(value, 32))
+            }
+
+            // Read original value
+            (bool success, bytes memory result) = _base.call(
+                abi.encodeWithSignature(string(abi.encodePacked(value, "()")))
+            );
```

Comment:
For the other two functions, this call is in one line. I'm fine either way but might be nice to keep it consistent changing this one or the other two.